Handle error fetching image from Met

```
docker run --rm -e SKIP_FETCH_CONFIG=true                 -e SKIP_FETCH_DATA=true                 -v $(pwd)/.:/opt/traject                 -v $(pwd)/../dlme-traject:/opt/traject/config                 -v $(pwd)/../dlme-metadata:/opt/traject/data                 -v $(pwd)/output:/opt/traject/output                 suldlss/dlme-transform:latest met
```

results in:
```
2019-02-27T15:57:10+00:00 ERROR Unexpected error on record <record #5 (data/met/ancient-near-east-art/data/met_museum_records.csv #5), output_id:met_321385>
    while executing (to_field "agg_preview" at config/met_csv_config.rb:40)

    Record: 74.51.4370,False,True,321385,Ancient Near Eastern Art,Stamp seal,Conoid seal,"",Late Cypriot III-Cypro-Geometric III,"","","","","","","","","","","","",ca. 12th–9th century B.C.,-1200,-800,"Steatite, brown","Seal Face; 1.61 x 1.46 cm
Height: 1.35 cm
String Hole: 0.25 cm","The Cesnola Collection, Purchased by subscription, 1874–76","","","","","",Cyprus (?),"","","","","",Stone-Stamp Seals,"",http://www.metmuseum.org/art/collection/search/321385,7/31/2017 8:00:01 AM,"Metropolitan Museum of Art, New York, NY"

    Exception: RuntimeError: Unexpected response type 'text/html' for http://www.metmuseum.org/api/Collection/additionalImages?crdId=321385
    /opt/traject/lib/dlme_utils.rb:24:in `fetch_json'

[ERROR] Unexpected response type 'text/html' for http://www.metmuseum.org/api/Collection/additionalImages?crdId=321385
2019-02-27T15:57:10+00:00 ERROR Unexpected error on record <record #1 (data/met/ancient-near-east-art/data/met_museum_records.csv #1), output_id:met_321381>
    while executing (to_field "agg_preview" at config/met_csv_config.rb:40)

    Record: 74.51.4366,False,True,321381,Ancient Near Eastern Art,Stamp seal,Quasi-pyramidal seal,Cypriot,Late Cypriot III,"","","","","","","","","","","","",ca. 12th–11th century B.C.,-1200,-1000,"Steatite, gray brown","Seal Face: 1.19 x 1.68 cm
Height: 1.78 cm
String Hole: 0.4 cm","The Cesnola Collection, Purchased by subscription, 1874–76","","","","","",Cyprus (?),"","","","","",Stone-Stamp Seals,"",http://www.metmuseum.org/art/collection/search/321381,7/31/2017 8:00:01 AM,"Metropolitan Museum of Art, New York, NY"

    Exception: RuntimeError: Unexpected response type 'text/html' for http://www.metmuseum.org/api/Collection/additionalImages?crdId=321381
    /opt/traject/lib/dlme_utils.rb:24:in `fetch_json'

D, [2019-02-27T15:57:10.144420 #7] DEBUG -- : DLME::Utils.fetch_json(http://www.metmuseum.org/api/Collection/additionalImages?crdId=321385) (433.3ms)
D, [2019-02-27T15:57:10.146086 #7] DEBUG -- : DLME::Utils.fetch_json(http://www.metmuseum.org/api/Collection/additionalImages?crdId=321381) (503.6ms)
```
where `http://www.metmuseum.org/api/Collection/additionalImages?crdId=321385` resolves to:
```
{
message: "An error has occurred."
}
```

--- a/lib/dlme_utils.rb
+++ b/lib/dlme_utils.rb
@@ -21,9 +21,9 @@ module DLME
         client.get uri
       end
       resp_content_type = resp.headers['content-type']
-      raise "Unexpected response type '#{resp_content_type}' for #{uri}" unless resp_content_type == 'application/json'
+      return ::JSON.parse(resp.body) if resp_content_type.start_with?('application/json')
 
-      ::JSON.parse(resp.body)
+      raise "Unexpected response type '#{resp_content_type}' for #{uri}"
     end
 
     def self.logger

--- a/lib/met_thumbnail_fetcher.rb
+++ b/lib/met_thumbnail_fetcher.rb
@@ -4,24 +4,24 @@ require 'dlme_utils'
 require 'active_support/core_ext/object/blank'
 
 # Grab a thumbnail URL from the Met web service
+# API docs at https://metmuseum.github.io/
 module MetThumbnailFetcher
   def self.fetch(ident)
     image_json = make_request(ident)
 
     return if image_json.blank?
 
-    unless image_json['results']
+    unless image_json['primaryImage']
       # Some records have null results
       DLME::Utils.logger.warn "No results found in #{ident}\n#{image_json}"
       return
     end
-    # Some records e.g. 321624, return empty results
-    result = image_json['results'].first
-    result['webImageUrl'] if result
+
+    image_json['primaryImage']
   end
 
   def self.make_request(id)
-    DLME::Utils.fetch_json("http://www.metmuseum.org/api/Collection/additionalImages?crdId=#{id}")
+    DLME::Utils.fetch_json("https://collectionapi.metmuseum.org/public/collection/v1/objects/#{id}")
   end
   private_class_method :make_request
 end

--- a/spec/lib/met_thumbnail_fetcher_spec.rb
+++ b/spec/lib/met_thumbnail_fetcher_spec.rb
@@ -10,21 +10,13 @@ RSpec.describe MetThumbnailFetcher do
 
     before do
       allow(DLME::Utils).to receive(:fetch_json)
-        .with('http://www.metmuseum.org/api/Collection/additionalImages?crdId=12312')
+        .with('https://collectionapi.metmuseum.org/public/collection/v1/objects/12312')
         .and_return(JSON.parse(json))
     end
 
     context 'when the results are empty' do
       let(:json) do
-        '{"results":[]}'
-      end
-
-      it { is_expected.to be nil }
-    end
-
-    context 'when the results are null' do
-      let(:json) do
-        '{"results":null}'
+        '{}'
       end
 
       it { is_expected.to be nil }
@@ -32,7 +24,7 @@ RSpec.describe MetThumbnailFetcher do
 
     context 'when there is a thumbnail' do
       let(:json) do
-        '{"results":[{"webImageUrl":"http://images.metmuseum.org/images/3"}]}'
+        '{"primaryImage":"http://images.metmuseum.org/images/3"}'
       end
 
       it { is_expected.to eq 'http://images.metmuseum.org/images/3' }

